staging/src/k8s.io/client-go/transport/round_trippers.go

```diff
@@ -87,6 +93,17 @@ type authProxyRoundTripper struct {
 	rt http.RoundTripper
 }
 
+// NewCacheRoundTripper creates a roundtripper that reads the ETag on
+// response headers and send the If-None-Match header on subsequent
+// corresponding requests.
+func NewCacheRoundTripper(cacheDir string, rt http.RoundTripper) http.RoundTripper {
+	d := diskv.New(diskv.Options{BasePath: cacheDir})
+	t := httpcache.NewTransport(diskcache.NewWithDiskv(d))
```

Comment:
> Sorry, I'm just learning about the ETag spec. Looks like when If-Modified-Since is used (which the httpcache library does), the server should return an HTTP 304

Yes,

> (which this PR's implementation should preserve)

It [doesn't](https://github.com/gregjones/httpcache/blob/master/httpcache.go#L187)